src/App.tsx

```diff
@@ -1,26 +1,97 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+// import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import * as todoService from './utils/helpers';
+import { Error } from './components/Errors';
+import { errorMessages } from './utils/const';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<todoService.Status>(
+    todoService.Status.all,
+  );
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoadingWhileDelete, setIsLoadingWhileDelete] =
+    useState<boolean>(false);
```

Comment:
In your case when I'm deleting any of the todos I see loader on all, that's not correct.
You can store loading todo ids in the array or set and indicate if this todo is currently loading.
```typescript
const [loadingTodoIds, setLoadingTodoIds] = useState([]);

// in the todo component just check
const isCurrentTodoLoading = loadingTodoIds.includes(todo.id)
```